File: platform-demos/C/samples/menubutton.js
#!/usr/bin/gjs

const _ = imports.gettext.gettext;
const Gio = imports.gi.Gio;
const GLib = imports.gi.GLib;
const Gtk = imports.gi.Gtk;
const Lang = imports.lang;

const Application = new Lang.Class ({
    Name: 'Application',

    //create the application
    _init: function () {
        this.application = new Gtk.Application ({
            application_id: 'org.example.myapp',
            flags: Gio.ApplicationFlags.FLAGS_NONE
        });

       //connect to 'activate' and 'startup' signals to the callback functions
       this.application.connect('activate', Lang.bind(this, this._onActivate));
       this.application.connect('startup', Lang.bind(this, this._onStartup));
    },

    //create the UI (in this case it's just the ApplicationWindow)
    _buildUI: function() {
        this._window = new Gtk.ApplicationWindow  ({ application: this.application,
                                                   window_position: Gtk.WindowPosition.CENTER,
                                                   title: "Welcome to GNOME" });
        this._window.set_default_size (600, 400);
        this.grid = new Gtk.Grid ();
        this._window.add (this.grid);
        
        
        this._menuButton = new Gtk.MenuButton();
        this.grid.attach (this._menuButton, 0, 0, 1, 1 );
        this.menu = new Gtk.Menu.new_from_model (this.menuModel);
        
        this.menu.show ();
        this._menuButton.set_menu (this.menu );
        this._menuButton.set_size_request (80, 35);
        this._menuButton.show ();
        
        this._window.show_all();
    },

    _showNew: function() {
    print ("New Menu. It doesn't do anything. It is only a demonstration.");
    },

    _showAbout: function() {
        print ("No AboutDialog here.  This is only a demonstration.");
    },
    


    //create the menu items and connect the signals to the callback functions.
    _initMenus: function() {
   
        let newAction = new Gio.SimpleAction ({ name: 'new' });
        newAction.connect('activate', Lang.bind(this,
            function() {
                this._showNew();
            }));
        this.application.add_action(newAction);

        let aboutAction = new Gio.SimpleAction ({ name: 'about' });
        aboutAction.connect('activate', Lang.bind(this,
            function() {
                this._showAbout();
            }));
        this.application.add_action(aboutAction);

        let quitAction = new Gio.SimpleAction ({ name: 'quit' });
        quitAction.connect('activate', Lang.bind(this,
            function() {
                this._window.destroy();
            }));
         this.application.add_action( quitAction );
         
        this.menuModel = new Gio.Menu();  
        
        this.menuItemNew = Gio.MenuItem.new( _("New"), 'app.new' );
        this.menuItemAbout = Gio.MenuItem.new( _("About"), 'app.about' );
        this.fileMenuItem = Gio.MenuItem.new( _("Other"), null);
        
        this.menuModel.append_item( this.menuItemNew );
        this.menuModel.append_item( this.menuItemAbout );
        
        //submenu
        this.subMenu = new Gio.Menu();
        this.fileMenuItem.set_submenu(this.subMenu);
        this.menuItemQuit = Gio.MenuItem.new( ("Quit"), 'app.quit' );
        this.subMenu.append_item( this.menuItemQuit );
        this.menuModel.append_item(this.fileMenuItem);
        
        
        
    },

    //callback function for 'activate' signal
    _onActivate: function() {
        this._window.present();
    },

    //callback function for 'startup' signal
    _onStartup: function() {
        //You must call _initMenus() before calling _buildUI().
        this._initMenus();
        this._buildUI();
    }
});

//run the application
let app = new Application ();
app.application.run (ARGV);

File: platform-demos/C/samples/menubutton.py
from gi.repository import Gtk
from gi.repository import Gio
import sys

class MyWindow(Gtk.ApplicationWindow):
    def __init__(self, app):
        Gtk.Window.__init__(self, title="Menubutton Example", application=app)
        self.set_default_size(600, 400)

        grid = Gtk.Grid()
        
        # a menubutton
        menubutton = Gtk.MenuButton()
        menubutton.set_size_request(80, 35)
        
        grid.attach(menubutton, 0, 0, 1, 1)
        
        # a menu with two actions
        menumodel = Gio.Menu()
        menumodel.append("New", "app.new")
        menumodel.append("About", "win.about")

        # a submenu with one action for the menu
        submenu = Gio.Menu()
        submenu.append("Quit", "app.quit")
        menumodel.append_submenu("Other", submenu)

        # the menu is set as the menu of the menubutton
        menubutton.set_menu_model(menumodel)

        # the action related to the window (about)
        about_action = Gio.SimpleAction.new("about", None)
        about_action.connect("activate", self.about_callback)
        self.add_action(about_action)
        
        self.add(grid)

    # callback for "about"
    def about_callback(self, action, parameter):
        print "You clicked \"About\""

class MyApplication(Gtk.Application):
    def __init__(self):
        Gtk.Application.__init__(self)

    def do_activate(self):
        win = MyWindow(self)
        win.show_all()

    def do_startup(self):
        Gtk.Application.do_startup(self)

        #  the actions related to the application
        new_action = Gio.SimpleAction.new("new", None)
        new_action.connect("activate", self.new_callback)
        self.add_action(new_action)
        
        quit_action = Gio.SimpleAction.new("quit", None)
        quit_action.connect("activate", self.quit_callback)
        self.add_action(quit_action)
    
    # callback functions for the actions related to the application
    def new_callback(self, action, parameter):
        print "You clicked \"New\""

    def quit_callback(self, action, parameter):
        print "You clicked \"Quit\""
        self.quit()

app = MyApplication()
exit_status = app.run(sys.argv)
sys.exit(exit_status)
